### ISSUE TITLE
[FEAT]: Videos over Nova's SMBJ start up nearly 3x slower than they do on Kodi's SMB

### ISSUE THREAD

**Issue by Pentaphon** [nova-video-player/aos-AVP#1056]
### Problem description

Videos over SMBJ start up nearly 3x slower than they do on Kodi

I notice one major difference when I compare the two players with a somewhat scientific test. 

**Nova's SMBJ starts videos in about 5.75-6 seconds.** 

**Kodi's SMB starts videos in about 2 seconds.**

I tried controlling for all variables in the steps below. I was able to reproduce the same result every single time. 

### Steps to reproduce the issue

1) Get fresh install of latest Nova (6.2.51) and Kodi (21 beta 2)
2) Play the same videos (in this case 1080p web-dl) from the same share, on the same network, using the same device. 
3) Get a stopwatch and time how long it takes for videos to open over SMBJ on Nova and SMB on Kodi. 

No matter how many times I kept testing and re-testing this, Kodi's SMB would start up faster than Nova's SMBJ. I went through about 20 videos of about the same bitrate and resolution so far and I would always get Kodi starting videos about 3x faster. 

Video would begin playing on Kodi's SMB in about 2 seconds.
Video would begin playing on Nova's SMBJ in about 5-6 seconds. 

Can anybody else reproduce this behavior? @courville this happens on both my TV devices, the Onn 4K 2021 and the Fire TV 4K stick.

### Expected behavior

You'd expect Nova's SMBJ to start videos as fast as Kodi does. Is there any way to speed up the video startup over SMBJ? 

This is not a huge issue but I do wonder why Kodi has closer to instant playback while Nova takes little while to start playing video under the exact same conditions. 

### Your phone/tablet/androidTV model

Onn 4K box (2021)

### Operating system version

Android 10

### Application version and app store

6.2.51

**Comment by courville** [comment]
Perhaps this is linked to buffering or subs uploading, I need to check. I would however perhaps classify this issue as "enhancement" and not "bug".

EDIT: to my knowledge also, kodi uses a custom C or C++ based SMB client and not a java one line smbj.

**Comment by Pentaphon** [comment]
> Perhaps this is linked to buffering or subs uploading, I need to check. I would however perhaps classify this issue as "enhancement" and not "bug".

Yes, my apologies. I should have put "enhancement" instead. I've changed the title. 

> EDIT: to my knowledge also, kodi uses a custom C or C++ based SMB client and not a java one line smbj.

Perhaps I should report this upstream to hopefully get some kind of optimization?

**Comment by courville** [comment]
> Perhaps I should report this upstream to hopefully get some kind of optimization?

I do not think so yet because we need to understand if this is not nova dependent first. Otherwise they will have no way to investigate or actionnable data to work with.
